spec/datadog/profiling/profiler_spec.rb

```diff
@@ -51,6 +87,20 @@
 
       shutdown!
     end
+
+    context 'when a crash tracker instance is provided' do
+      let(:optional_crashtracker) { instance_double(Datadog::Profiling::Crashtracker) }
+
+      it 'signals the crash tracker to stop, after other components have stopped' do
+        expect(worker).to receive(:stop).ordered
+        allow(scheduler).to receive(:enabled=)
+        expect(scheduler).to receive(:stop).ordered
+
+        expect(optional_crashtracker).to receive(:stop).ordered
+
+        shutdown!
+      end
+    end
   end
 
   describe 'Component failure handling' do
```

Comment:
If we want to keep the crashtracker alive in these situations (which I think makes sense), may want to add a test for this case to make sure we don't inadvertently remove it later on.